src/services/notehub/NotehubAttributeStore.ts

```diff
@@ -0,0 +1,27 @@
+import { AttributeStore } from "../AttributeStore";
+import { NotehubAccessor } from "./NotehubAccessor";
+
+export class NotehubAttributeStore implements AttributeStore {
+  constructor(private accessor: NotehubAccessor) {}
+
+  async updateSensorName(gatewayUID: string, macAddress: string, name: string) {
+    const { body } = await this.accessor.getConfig(gatewayUID, macAddress);
+    body.name = name;
+    await this.accessor.setConfig(gatewayUID, macAddress, body);
+    return true;
+  }
+
+  async updateSensorLocation(
+    gatewayUID: string,
+    macAddress: string,
+    loc: string
+  ) {
+    const { body } = await this.accessor.getConfig(gatewayUID, macAddress);
```

Comment:
Are we not worried about concurrent updates?  It's very much an edge case so we can of course live with this, but do we have any other possible approach?  I'm guessing notehub doesn't have a read-modify API for db files?

Reply:
We can implement that when we get to it. For now, I'm aiming to implement the [ant design pattern 'make it direct'](https://ant.design/docs/spec/direct) as discussed with Michelle, so each item is edited right when you click on its edit pencil icon and hit enter.

Reply:
What I mean is if two people were editing the same sensor details at the same time, one of them would lose their edits, even if one edits the name and the other the location, due to the concurrent non-atomic read/modify/write of the entire `config.db` record.   It's not a big concern - so feel free to punt and ignore for now,  but I imagine it's a useful pattern for notehub to implement to detect concurrent updates, e.g.  a compare and exchange to implement atomic updates optimistically (with the possibility of failure if the current data doesn't meet the compare condition because it has been updated in the meantime.)

Reply:
Oh yeah that _concurrency_. I'm not too worried yet. Last writer wins isn't too unusual in the cloud. Plus, Notehub is a data pump not a database. :wink: